isb_web/sqlmodel_database.py

```diff
@@ -306,6 +361,38 @@ def save_or_update_thing(session: Session, thing: Thing):
                 )
 
 
+def all_thing_identifier_objects(
+    session: Session, min_id: int, batch_size: int
+) -> list[ThingIdentifier]:
+    thing_identifiers_select = (
+        select(ThingIdentifier)
+        .filter(ThingIdentifier.thing_id >= min_id)
+        .order_by(ThingIdentifier.thing_id.asc())
+        .limit(batch_size)
+    )
+    thing_identifiers = session.exec(thing_identifiers_select).all()
+    return thing_identifiers
+
+
+def things_with_null_identifiers(session: Session) -> list[Thing]:
+    # noinspection PyComparisonWithNone
+    things_select = select(Thing).filter(Thing.identifiers == None)
+    things = session.exec(things_select).all()
+    return things
```

Comment:
I believe I mentioned previously that it looked like some records were inexplicably missing identifiers -- this finds those records.